src/pytorch_lightning/utilities/rank_zero.py

```diff
@@ -97,3 +100,24 @@ class LightningDeprecationWarning(DeprecationWarning):
 
 
 rank_zero_deprecation = partial(rank_zero_warn, category=LightningDeprecationWarning)
+
+
+def _rank_prefixed_message(
+    message: str, trainer: Optional["pl.Trainer"] = None, log_rank_zero_only: bool = False
+) -> Optional[str]:
+    if trainer:
+        # ignore logging in non-zero ranks if `log_rank_zero_only` flag is enabled
+        if log_rank_zero_only and trainer.global_rank != 0:
+            return None
+        if trainer.world_size > 1:
+            # specify the rank of the process being logged
+            return f"[rank: {trainer.global_rank}] {message}"
+    else:
+        rank = _get_rank()  # TODO: this returns the local rank not global. Inconsistent with the trainer case
```

Comment:
Thoughts on this TODO @awaelchli? Early stopping used global rank but `_get_rank` uses local rank

Reply:
Where does this TODO come from? `_get_rank()` returns global rank. It is consistent with the trainer, except for custom cluster environments.